string connect
---
string returncode_t